server.go

```diff
@@ -85,6 +85,50 @@ var (
 		version.Patch)
 )
 
+// onionAddr implements the net.Addr interface and represents a tor address.
+type onionAddr struct {
```

Comment:
What's the reasoning for adding this?  It isn't used anywhere.

Also, if we want to keep it, the two address types should be made consistent.  This one is a pointer and `simpleAddr` isn't.